### PR TITLE
Hardening

### PR DESCRIPTION
**Pull request by teryk**
👨‍🏭 🚧 👷

### CODE REVIEW

File: server/sonar-db-dao/src/main/java/org/sonar/db/metric/MetricDao.java [hunk 1]
```diff
@@ -81,7 +82,7 @@ public int countEnabled(DbSession session, @Nullable Boolean isCustom) {
   }
 
   public MetricDto insert(DbSession session, MetricDto dto) {
-    mapper(session).insert(dto);
+    mapper(session).insert(validate(dto));
```
Comment (julienlancelot) [inline server/sonar-db-dao/src/main/java/org/sonar/db/metric/MetricDao.java]: AFAIK, It's not the responsability of DAO to validate dto. This should be in a higher layer, a XXXUpdater or the WS

Comment (julienlancelot) [inline server/sonar-db-dao/src/main/java/org/sonar/db/metric/MetricDao.java]: Or maybe in the dto setters

File: server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndex.java [hunk 2]
```diff
@@ -348,15 +348,16 @@ private static QueryBuilder toValueQuery(MetricCriterion criterion) {
     }
   }
 
-  public List<String> searchTags(@Nullable String textQuery, int pageSize) {
-    checkArgument(pageSize <= 100, "Page size must be lower than or equals to " + 100);
-    if (pageSize == 0) {
+  public List<String> searchTags(@Nullable String textQuery, int size) {
+    int maxPageSize = 500;
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndex.java]: Please don't forget to add this WS in the ticket

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java [hunk 3]
```diff
@@ -177,6 +172,160 @@ public IssueIndex(EsClient client, System2 system, UserSession userSession, Auth
     this.sorting.addDefault(IssueIndexDefinition.FIELD_ISSUE_KEY);
   }
 
+  /**
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java]: Not sure it was a good idea to reformat the class in this commit as it's now super hard to see what has changed...

File: server/sonar-db-dao/src/main/java/org/sonar/db/metric/MetricDtoValidator.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+public class MetricDtoValidator {
```
Comment (julienlancelot) [inline server/sonar-db-dao/src/main/java/org/sonar/db/metric/MetricDtoValidator.java]: The JIRA ticket should contains really what have you done, as the ticket is only about description, and if my understanding is correct you have added more checks than only on description

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java (hunk 3 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java]: indeed 😭

**Pull request merged**